### PR TITLE
[GO]: OTel API operation naming: Unskipped / Fixed the tests in accordance to OTel attributes remapping

### PR DESCRIPTION
**Pull request by dianashevchenko**
## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/apm_tracing_e2e/test_otel.py [hunk 1]
```diff
@@ -26,57 +27,63 @@ def test_datadog_otel_span(self):
...
+        assert handler_span["resource"] == "testOperation"
         assert handler_span.get("parentID") == roundtrip_span.get("spanID")
 
         # Assert the spans received from the backend!
         spans = interfaces.backend.assert_request_spans_exist(self.req, query_filter="")
         assert 3 == len(spans), _assert_msg(3, len(spans))
 
 
-def _get_span(spans, span_name):
+def _get_span(spans, span_name, resource_name):
```
Comment (robertomonteromiguel) [inline tests/apm_tracing_e2e/test_otel.py:82]: Perhaps, we can split this method in _get_span_by_name and _get_span_by_resource or something like this, in order to easily read the code. It's a proposal....

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
There is a test failure on ci
There are conflicts with the main branch

File: tests/apm_tracing_e2e/test_otel.py (hunk 1 above)
Comment (dianashevchenko) [reply tests/apm_tracing_e2e/test_otel.py]: I also merged the main

**Review by katiehockman — APPROVED** [review]

**Comment by cbeauchesne** [comment]
@dianashevchenko please ping me if you need any help to solve the egg-chicken

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**